File: Python Programs/unit2/ArrayInsertion.py
from __future__ import print_function


def Array_Insertion(array, n, element, pos):
    array.append(0)  # Increasing the size of the array
    print("No. of element : %d,pos : %d" % (n, pos))
    for i in reversed(range(pos, n)):
        array[i + 1] = array[i]  # SWAPPING
    print("element : %d,pos : %d" % (element, pos))
    if pos > n:
        array[pos - 1] = element
    else:
        array[pos] = element  # INSERTION OF ELEMENTS

    print("array[pos] = %d, n=%d" % (array[pos], n))
    n += 1
    print("****after insertion****")
    for i in range(0, n):
        print("%d\t" % array[i])
    print("no of elements=%d" % n)


arr = []
print("Enter the number of elements in an array")
n = int(input())
print("Enter the elements")
for i in range(0, n):
    arr.append(int(input()))
print("Elements in the array")
for i in arr:
    print("%d" % i)

print("no of elements=%d\n Enter the element to be inserted" % n)
e = int(input())
print("enter the position of the element\n")
pos = int(input())
if(pos > 100):
    print("Insertion not possible")
else:
    Array_Insertion(arr, n, e, pos)

File: Java Programs/Chapter 2/Array_Insertion/src/array_insertion/Array_Insertion.java

package array_insertion;

//Program to insert an element in an array.
//The elements of the array maybe of any datatype.
import java.util.*;

public class Array_Insertion
{
	public static void main(String args[])
	{
		int n=10;
		Object[] A=new Object[n];
		Object element;
		int pos,noElements,i;
		
		Scanner oin=new Scanner(System.in);
		System.out.println("Enter the number of elements in the array");
		noElements=Integer.parseInt(oin.nextLine());
			
		System.out.println("Enter the elements of the array:");
		for(i=1;i<=noElements;i++)
		{
			A[i]=(Object)oin.nextLine();
		}	

		for(i=1;i<=noElements;i++)
		{
			System.out.print(""+A[i]+"	");
		}
		System.out.print("\nEnter the element to be inserted :");
		element=(Object)oin.nextLine();
		System.out.print("\nEnter the position where the element is to be inserted :");
		pos=Integer.parseInt(oin.nextLine());
		
		if(noElements==n || (pos>noElements+1))
		{
			System.out.println("Element cannot be inserted");
		}
		else
		{
			for(i=noElements;i>=pos;i--)
			{
				A[i+1]=A[i];
			}
			A[pos]=element;
			noElements=noElements+1;
		}
		System.out.println("\nNew Array");
		
		for(i=1;i<=noElements;i++)
		{
			System.out.print(""+A[i]+" | ");
		}
		
	}
}